fedimint-api/src/db/mod.rs

```diff
@@ -680,6 +680,40 @@ where
     }
 }
 
+/// This is a helper macro that generates the implementations of
+/// DatabaseKeyPrefixConst necessary for reading/writing to the
+/// database and fetching by prefix.
+///
+/// - `key`: This is the type of struct that will be used as the key into the database.
+/// - `value`: This is the type of struct that will be used as the value into the database.
+/// - `prefix`: Enum expression that is represented as a `u8` that is the prefix prepended to this key.
+/// - `key_prefix`: Optional type of struct that can be passed multiple times if necessary. These will be used
```

Comment:
assuming we called the `prefix` -> `prefix_db`, then we could call the `key_prefix`  `prefix_query` or similar. As it's used to *query* for a specific key prefixed first with the `prefix_db` itself.  

it's `<prefix_db><prefix_query><rest_of_key_bytes>`, right?

Reply:
`prefix_query` is just the struct that holds the prefix byte and defines the types for the `Key` and `Value`. I enabled auto-merge so this already got merged, but we can rename them in a follow up if you'd like.

On disk I believe it looks something like this:
`<Global Module Prefix Byte><Module Prefix Byte Upper><Module Prefix Byte Lower><prefix_db><rest of key bytes>` 

The global module prefix byte and module prefix bytes are appended even before `prefix_db` when the key belongs to a module.

Reply:
so just for completeness as we talked off-thread. the `<module byte upper>`  and `<module byte lower>` is the module instance ID, which is a `u16` so 2 bytes.  The global module prefix is currently defined as `0xff`. 
so
`<Global Module Prefix><Module Instance Prefix><prefix_db><key_bytes>`

thanks @m1sterc001guy 

@ others: also take a look at [the DB documentation](https://github.com/fedimint/fedimint/blob/master/docs/database.md#server-db-layout)